packages/cli/src/apiErrorHandling.ts

```diff
@@ -0,0 +1,29 @@
+import debuggers, { DebugNamespaces } from './debug';
+import { NonVerboseError } from './errors';
+
+const debug = debuggers[DebugNamespaces.UPLOAD];
+
+type ApiError = {
+  message: string;
+};
+
+export class InvalidSecretError extends NonVerboseError {
+  constructor() {
+    super('Your secret is not valid, did you misspell it?');
+  }
+}
+
+export function handleApiError(err) {
+  if (!err.response) {
+    return;
+  }
+
+  debug('API error data', err.response?.data);
+  if (err.response.status === 400 && hasInvalidSecretError(err.response.data.errors)) {
+    throw new InvalidSecretError();
+  }
```

Comment:
whats with other errors? dont we throw sth generic as well?

Reply:
No, because we might want to handle them afterwards. This is just to handle api generic errors that might happen in different APIs